packages/collection/src/index.ts

```diff
@@ -625,14 +625,16 @@ export class Collection<K, V> extends Map<K, V> {
 	}
 
 	/**
-	 * The intersect method returns a new structure containing items where the keys are present in both original structures.
+	 * The intersect method returns a new structure containing items where the keys and values are present in both original structures.
 	 *
 	 * @param other The other Collection to filter against
 	 */
-	public intersect<T>(other: Collection<K, T>): Collection<K, T> {
-		const coll = new this.constructor[Symbol.species]<K, T>();
+	public intersect<T>(other: Collection<K, T>): Collection<K, T | V> {
+		const coll = new this.constructor[Symbol.species]<K, T | V>();
 		for (const [k, v] of other) {
-			if (this.has(k)) coll.set(k, v);
+			if (this.has(k) && Object.is(v, other.get(k))) {
```

Comment:
Did you mean to do this instead?
```suggestion
			if (this.has(k) && Object.is(v, this.get(k))) {
```

`other.get(k)` (while iterating `other`) equals to `v`, so this statement will never be false.